pkgs/servers/nextcloud/packages/apps/memories.nix

```diff
@@ -0,0 +1,89 @@
+{
+  stdenv,
+  buildGoModule,
+  exiftool,
+  fetchurl,
+  ffmpeg,
+  fetchFromGitHub,
+  lib,
+
+  ncVersion,
+}:
+let
+  latestVersionForNc = {
+    "31" = latestVersionForNc."29";
+    "30" = latestVersionForNc."29";
+    "29" = {
+      version = "7.5.2";
+      appHash = "sha256-BfxJDCGsiRJrZWkNJSQF3rSFm/G3zzQn7C6DCETSzw4=";
+      srcHash = "sha256-imBO/64NW5MiozpufbMRcTI9WCaN8grnHlVo+fsUNlU=";
+    };
+  };
+  currentVersionInfo =
+    latestVersionForNc.${ncVersion}
+      or (throw "memories currently does not support nextcloud version ${ncVersion}");
+
+  commonMeta = with lib; {
+    homepage = "https://apps.nextcloud.com/apps/memories";
+    changelog = "https://github.com/pulsejet/memories/blob/v${currentVersionInfo.version}/CHANGELOG.md";
+    license = licenses.agpl3Only;
+    maintainers = with maintainers; [ SuperSandro2000 ];
+  };
+
+  go-vod = buildGoModule rec {
+    pname = "go-vod";
+    inherit (currentVersionInfo) version;
+
+    src = fetchFromGitHub {
+      owner = "pulsejet";
+      repo = "memories";
+      tag = "v${version}";
+      hash = currentVersionInfo.srcHash;
+    };
+
+    sourceRoot = "${src.name}/go-vod";
+
+    vendorHash = null;
+
+    meta = commonMeta // {
+      description = "Extremely minimal on-demand video transcoding server in go";
+      mainProgram = "go-vod";
+    };
+  };
+in
+stdenv.mkDerivation rec {
+  pname = "nextcloud-app-memories";
+  inherit (currentVersionInfo) version;
+
+  src = fetchurl {
+    url = "https://github.com/pulsejet/memories/releases/download/v${version}/memories.tar.gz";
+    hash = currentVersionInfo.appHash;
+  };
+
+  postPatch = ''
+    ln -s ${lib.getExe go-vod} bin-ext/go-vod
+    for bin in bin-ext/{go-vod-aarch64,go-vod-amd64}; do
+      ln -sf go-vod $bin
+    done
+
+    for bin in bin-ext/{exiftool-aarch64-glibc,exiftool-aarch64-musl,exiftool-amd64-glibc,exiftool-amd64-musl}; do
```

Comment:
otoh... can't we just patch the paths in the code directly?

Reply:
I think that might be a good idea. 

The files below are good targets I think:

https://github.com/pulsejet/memories/blob/master/lib/Settings/SystemConfig.php
https://github.com/pulsejet/memories/blob/master/lib/Service/BinExt.php

Reply:
We would need to half rewrite the entirety of [BinExt](https://github.com/pulsejet/memories/blob/master/lib/Service/BinExt.php) which is like 400 loc and I am not looking forward to rebasing those patches or to test things especially with php where compile time checks don't really exist.